cli/wordpress-server-child.ts

```diff
@@ -0,0 +1,187 @@
+	return str.replace( /\\/g, '\\\\' ).replace( /'/g, "\\'" );
+}
+
+async function setAdminPassword( server: RunCLIServer, adminPassword: string ): Promise< void > {
+	await server.playground.request( {
+		url: '/?studio-admin-api',
+		method: 'POST',
+		body: {
+			action: 'set_admin_password',
+			password: escapePhpString( adminPassword ),
+		},
+	} );
+}
+
+async function startServer( config: ServerConfig ): Promise< void > {
+	if ( server ) {
+		console.log( `[WordPress Server Child] Server already running for site ${ config.siteId }` );
+		return;
+	}
+
+	try {
+		const hasWordPress = isWordPressDirectory( config.sitePath );
+
+		const [ studioMuPluginsHostPath, loaderMuPluginHostPath ] = await getMuPlugins( {
+			isWpAutoUpdating: config.isWpAutoUpdating,
+		} );
+
+		const defaultConstants = {
+			WP_SQLITE_AST_DRIVER: true,
+		};
+
+		const mounts = [
+			{
+				hostPath: config.sitePath,
+				vfsPath: '/wordpress',
+			},
+			{
+				hostPath: studioMuPluginsHostPath,
+				vfsPath: '/internal/studio/mu-plugins',
+			},
+			{
+				hostPath: loaderMuPluginHostPath,
+				vfsPath: '/internal/shared/mu-plugins/99-studio-loader.php',
+			},
+		];
+
+		const args: RunCLIArgs = {
+			command: 'server',
+			internalCookieStore: false,
+			login: false,
+			followSymlinks: true,
+			skipSqliteSetup: true,
+			port: config.port,
+			'mount-before-install': mounts,
+			'site-url': config.absoluteUrl || `http://localhost:${ config.port }`,
+			blueprint: config.blueprint || {},
+			wordpressInstallMode: 'download-and-install',
+		};
+
+		if ( hasWordPress ) {
+			args.wordpressInstallMode = 'install-from-existing-files-if-needed';
+		}
+
+		if ( config.phpVersion ) {
+			args.php = config.phpVersion as SupportedPHPVersion;
+		}
+
+		if ( config.wpVersion ) {
+			if ( isWordPressDevVersion( config.wpVersion ) ) {
+				args.wp = 'nightly';
+			} else {
+				args.wp = config.wpVersion;
+			}
+		}
+
+		args.blueprint.constants = { ...args.blueprint.constants, ...defaultConstants };
+		const server = await runCLI( args );
+
+		if ( ! server ) {
+			throw new Error( 'Failed to start server: runCLI returned void' );
+		}
+
+		if ( config.adminPassword ) {
+			await setAdminPassword( server, config.adminPassword );
+		}
+	} catch ( error ) {
+		server = null;
+		console.error( `[WordPress Server Child] Failed to start server:`, error );
+		throw error;
+	}
+}
+
+if ( process.send ) {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	process.on( 'message', async ( packet: any ) => {
+		try {
+			const message: Message = packet.type === 'process:msg' && packet.data ? packet.data : packet;
+
+			const validationResult = MessageSchema.safeParse( message );
+			if ( ! validationResult.success ) {
+				throw new Error(
+					`Invalid message format: ${ validationResult.error.errors
+						.map( ( e ) => e.message )
+						.join( ', ' ) }`
+				);
+			}
+
+			const validMessage = validationResult.data;
```

Comment:
```suggestion
	process.on( 'message', async ( packet: unknown ) => {
		try {
			const validMessage = MessageSchema.parse( packet );
```
We can simplify this now that we have zod validation.